src/lib.rs

```diff
@@ -140,78 +140,108 @@ impl CursorThemeIml {
-            xcursor_path.push(PathBuf::from("/usr/share/cursors/xorg-x11"));
+    if let Some(xdg_data_home) = environment.xdg_data_home {
+        paths.extend(expand_home_dir(PathBuf::from(xdg_data_home), home_dir));
+    } else if let Some(home_dir) = home_dir {
+        paths.push(home_dir.join(".local/share/icons"))
+    }
 
-            xcursor_path
-        }
-    };
+    if let Some(home_dir) = home_dir {
+        paths.push(home_dir.join(".icons"));
+    }
 
-    let homedir = env::var("HOME");
+    if let Some(xdg_data_dirs) = environment.xdg_data_dirs {
+        paths.extend(parse_to_icons_dir_list(&xdg_data_dirs, home_dir));
+    } else {
+        paths.push(PathBuf::from("/usr/local/share/icons"));
+        paths.push(PathBuf::from("/usr/share/icons"));
+    }
 
-    xcursor_path
-        .into_iter()
-        .filter_map(|dir| {
-            // Replace `~` in a path with `$HOME` for compatibility with other libs.
-            let mut expaned_dir = PathBuf::new();
+    paths.push(PathBuf::from("/usr/share/pixmaps"));
 
-            for component in dir.iter() {
-                if component == "~" {
-                    let homedir = match homedir.as_ref() {
-                        Ok(homedir) => homedir.clone(),
-                        Err(_) => return None,
-                    };
+    if let Some(home_dir) = home_dir {
+        paths.push(home_dir.join(".cursors"));
+    }
 
-                    expaned_dir.push(homedir);
-                } else {
-                    expaned_dir.push(component);
-                }
-            }
+    paths.push(PathBuf::from("/usr/share/cursors/xorg-x11"));
+
+    paths
+}
 
-            Some(expaned_dir)
+/// Parses colon separated path lists, skipping empty elements and expanding home dirs in paths that
+/// begin with `~/`. Adds `icons` dir to every path.
+fn parse_to_icons_dir_list(path_list: &str, home_dir: Option<&Path>) -> Vec<PathBuf> {
+    path_list
+        .split(':')
+        .flat_map(|entry| {
+            if entry.is_empty() {
+                return None;
+            }
+            let mut entry = expand_home_dir(PathBuf::from(entry), home_dir)?;
+            entry.push("icons");
+            Some(entry)
         })
         .collect()
 }
 
+/// If the first component of the path is `~`, replaces it with the home dir. If no home dir is
+/// present, returns `None`.
+fn expand_home_dir(path: PathBuf, home_dir: Option<&Path>) -> Option<PathBuf> {
+    let mut components = path.iter();
+    if let Some(first_component) = components.next() {
+        if first_component == "~" {
+            if let Some(home_dir) = home_dir {
+                let mut path = home_dir.to_path_buf();
+                for component in components {
+                    path.push(component);
+                }
+                return Some(path);
+            } else {
+                return None;
+            }
+        }
+    }
+    return Some(path);
```

Comment:
```suggestion
    Some(path)
```

Clippy complains about using `return` in this way